### DISCUSSION TITLE
Won't compile on fresh Julia install, complains about dependencies

### DISCUSSION THREAD

**Discussion by erin-atkinson** [CliMA/Oceananigans.jl#2895] [General]
Hi, I just installed Oceananigans, it gives this error when precompiling:

(@v1.8) pkg> precompile
Precompiling project...
  ✗ Oceananigans
  0 dependencies successfully precompiled in 4 seconds. 146 already precompiled.

ERROR: The following 1 direct dependency failed to precompile:

Oceananigans [9e8cae18-63c1-5223-a75c-80ca9d6e9a09]

Failed to precompile Oceananigans [9e8cae18-63c1-5223-a75c-80ca9d6e9a09] to /home/yolk/.julia/compiled/v1.8/Oceananigans/jl_lpMZRF.
ERROR: LoadError: ArgumentError: Package Oceananigans does not have StaticArrays in its dependencies:
- You may have a partially installed environment. Try `Pkg.instantiate()`
  to ensure all packages in the environment are installed.
- Or, if you have Oceananigans checked out for development and have
  added StaticArrays as a dependency but haven't updated your primary
  environment's manifest file, try `Pkg.resolve()`.
- Otherwise you may need to report an issue with Oceananigans
Stacktrace:
 [1] macro expansion
   @ ./loading.jl:1167 [inlined]
 [2] macro expansion
   @ ./lock.jl:223 [inlined]
 [3] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1144
 [4] include
   @ ./Base.jl:419 [inlined]
 [5] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
   @ Base ./loading.jl:1554
 [6] top-level scope
   @ stdin:1
in expression starting at /home/yolk/.julia/packages/Oceananigans/KClIg/src/Oceananigans.jl:1
in expression starting at stdin:1

Tried `instantiate` and `resolve`. Here is the version info.

Julia Version 1.8.5
Commit 17cfb8e65ea (2023-01-08 06:45 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 6 × AMD Ryzen 5 4500U with Radeon Graphics
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, znver2)
  Threads: 1 on 6 virtual cores

**Comment by glwagner** [comment]
Is this on a cluster? I've seen this before but no idea where it comes from. I've also gotten mysterious and similar errors regarding `GPUArrays` when trying to install `CUDA`.

One thing you might try is to add `StaticArrays` to your global environment, and the install / precompile Oceananigans. Not sure if that will help.

**Reply by glwagner** [reply]
Oddly, `StaticArrays` is not a direct dependency of Oceananigans:

https://github.com/CliMA/Oceananigans.jl/blob/c21d5d6ac95e762cac85154c8f3c4de63631fca7/Project.toml#L6-L39

and is not used in the Oceananigans source code. However, it is a dependency of ~~`CUDAKernels`~~ a few packages we depend on:

https://github.com/CliMA/Oceananigans.jl/blob/c9c6e2b2574b87b523b6df960650f4a7f66d0b44/Manifest.toml#L318-L319

https://github.com/CliMA/Oceananigans.jl/blob/c9c6e2b2574b87b523b6df960650f4a7f66d0b44/Manifest.toml#L82-L83

https://github.com/CliMA/Oceananigans.jl/blob/c9c6e2b2574b87b523b6df960650f4a7f66d0b44/Manifest.toml#L506-L507

https://github.com/CliMA/Oceananigans.jl/blob/c9c6e2b2574b87b523b6df960650f4a7f66d0b44/Manifest.toml#L576-L577

**Comment by glwagner** [comment]
I wonder if this is related to https://github.com/JuliaLang/julia/issues/42873

**Comment by erin-atkinson** [comment]
Hm.. I changed my setup a bit to avoid it (this was on Windows Subsystem for Linux, I'm just using it on Windows now). Luckily it works fine on the cluster I'm using later. If it occurs again I'll try adding the offender to the global environment but for now I'm okay.

**Reply by glwagner** [reply]
Sounds about right for this ephemeral issue! Maybe it'll get solved via this discussion sooner or later
